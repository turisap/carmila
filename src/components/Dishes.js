/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

export default class Dishes extends React.Component{
    render() {
        return (
            <h2>Dishes</h2>
        )
    }
}