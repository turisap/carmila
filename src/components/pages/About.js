/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const About = () => (
    <div>
        <h1>About page</h1>
    </div>
);

export default About;