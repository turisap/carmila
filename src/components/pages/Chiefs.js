/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const Chiefs = () => (
    <div className="container">
        <div className="row">
            <div className="col-md-6">

            </div>
        </div>
    </div>
);

export default Chiefs;