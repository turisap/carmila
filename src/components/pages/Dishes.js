/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import { connect } from 'react-redux';
import DishItem from './../DishItem';
import visibleDishes from '../../selectors/visibleDishes';
import getRowsOfDishes from '../../adapters/chunk';
import DishesFilterForm from '../DishesFilterForm';
import { addItem, removeItem, fillCredentials } from '../../actions/orders';


export class Dishes extends React.Component {
    handleAddButton = (item) => {
        this.props.addItemToOrder(item)
    };

    getAddedTitles = () => {
        return [...this.props.items].map(item => item.title);
    };

    render(){
        return (
            <div>
                <section className="dishesFilterForm">
                    <div className="container-fluid">
                        <div className="row">
                            <div className="col-md-12">
                                <DishesFilterForm/>
                            </div>
                        </div>
                    </div>
                </section>
                <section className="dishList">
                    <div className="container-fluid">
                        {getRowsOfDishes(3, this.props.dishes, false, this.handleAddButton, this.getAddedTitles())}
                    </div>
                </section>

            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    dishes : visibleDishes(state.filter.filterBy, state.filter.text),
    items  : state.orders.items
});

const mapDispatchToProps = (dispatch, props) => ({
    addItemToOrder : (item) => dispatch(addItem(item)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Dishes);