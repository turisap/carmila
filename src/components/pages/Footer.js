/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {NavLink} from 'react-router-dom';

const Footer = () => (
    <div className="footer__fix">

        <footer>
            <ul className="footer__links">
                <li><NavLink className="footer__link text-center" to="/about">About Us</NavLink></li>
                <li><NavLink className="footer__link text-center" to="#">Sign Up For News</NavLink></li>
            </ul>
            <ul className="footer__info">
                <li>
                    <p id="footer__info">Our office is located 24, Rd No 2 Krong Battambang.
                        We are open 8a.m. to 5p.m Monday to Friday.</p>
                </li>
            </ul>
        </footer>
    </div>
);

export default Footer;