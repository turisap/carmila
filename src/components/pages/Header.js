/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';

export const Header = (props) => (
    <div>
        <nav className="navbar navbar-expand-lg navbar-light bg-light navbar-right">
            <NavLink className="navbar-brand" to="/">Caldira</NavLink>
            <div className="collapse navbar-collapse" id="navbarColor03">
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/">Home</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/dishes">Menu</NavLink>
                    </li>
                    {props.haveItemsInOrder &&
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/order">Order</NavLink>
                    </li>
                    }
                </ul>
            </div>
        </nav>
    </div>
);

const mapStateToProps = (state) => ({
    haveItemsInOrder : (state.orders.items.length > 0)
});

export default connect(mapStateToProps)(Header);