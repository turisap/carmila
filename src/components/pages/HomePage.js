/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';


const HomePage = () => (
<div>
    <section id="bestFood">
        <div className="container-fluid bestFood__container">
            <div className="row">
                <div className="col-md-6 offset-md-3">
                    <h1 className="text-center bestFood__title">Find the best food in</h1>
                    <h1 className="text-center bestFood__title">Battambang</h1>
                </div>
            </div>
        </div>
    </section>
    <section id="homePagePhotos">
        <div className="container-fluid">
            <div className="row">
                <div className="col-md-4">
                    <img className="homePagePhotos_image homePagePhotos_main" src={"./img/pages/fire.jpg"}/>
                </div>
                <div className="col-md-4">
                    <div className="homePagePhotos_cropper">
                        <img className="homePagePhotos_image"  src={"./img/pages/dessert.jpg"}/>
                        <img className="homePagePhotos_image"  src={"./img/pages/pancakes.jpg"}/>
                    </div>
                </div>
                <div className="col-md-4">
                    <div className="homePagePhotos_cropper">
                        <img className="homePagePhotos_image"  src={"./img/pages/pizza_home.jpg"}/>
                        <img className="homePagePhotos_image"  src={"./img/pages/salami.jpg"}/>
                    </div>
                </div>
            </div>
        </div>
    </section>
</div>
);

export default HomePage;