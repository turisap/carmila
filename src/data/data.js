/**
 * Created by HP on 10-Dec-17.
 */
export const categories = [
    'PIZZA',
    'PASTA',
    'DESSERTS',
];

export const dishes = [
    {
        category: categories[0],
        title: '4 Cheeses',
        price: 5.5,
        description : 'Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut ',
        path : 'img/dishes/4chesses.jpg',
        amount : undefined
    },
    {
        category: categories[0],
        title: 'Margherita',
        price: 7,
        description : 'Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus.',
        path : 'img/dishes/margarita.jpg',
        amount : undefined
    },
    {
        category: categories[0],
        title: 'Al Taglio',
        price: 6.5,
        description : 'Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut ',
        path : 'img/dishes/altagila.jpg',
        amount : undefined
    },
    {
        category: categories[0],
        title: 'Quattro',
        price: 6,
        description : 'At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.',
        path : 'img/dishes/quatro.jpg',
        amount : undefined
    },
    {
        category: categories[1],
        title: 'Ravioli',
        price: 6.5,
        description : 'Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.',
        path : 'img/dishes/ravioli.jpg',
        amount : undefined
    },
    {
        category: categories[1],
        title: 'Fettuccini',
        price: 6.5,
        description : 'At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.',
        path : 'img/dishes/fetuccini.jpeg',
        amount : undefined
    },
    {
        category: categories[1],
        title: 'Rigatoni',
        price: 7,
        description : 'Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem.',
        path : 'img/dishes/rigatoni.jpg',
        amount : undefined
    },
    {
        category: categories[1],
        title: 'Linguine',
        price: 5.5,
        description : 'Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.',
        path : 'img/dishes/linguine.jpg',
        amount : undefined
    },
    {
        category: categories[1],
        title: 'Penne',
        price: 7.5,
        description : 'At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.',
        path : 'img/dishes/penne.jpg',
        amount : undefined
    },
    {
        category: categories[2],
        title: 'Serradura',
        price: 4,
        description : 'Maecenas porttitor eros nunc, vitae mollis tellus convallis sollicitudin. Integer fermentum turpis quis nibh pretium, et mattis quam fringilla. Maecenas dictum, magna ac tempor dictum, nisl ex laoreet nisi, nec porttitor turpis elit non leo. Pellentesque at elementum eros, a viverra velit. Nulla luctus libero vitae urna ultrices pulvinar.',
        path : 'img/dishes/serradura.jpg',
        amount : undefined
    },
    {
        category: categories[2],
        title: 'Clafoutis',
        price: 4.5,
        description : 'Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.',
        path : 'img/dishes/clafouits.jpg',
        amount : undefined
    },
    {
        category: categories[2],
        title: 'Cremeschnitte',
        price: 4,
        description : 'Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem.',
        path : 'img/dishes/cremeshnitte.jpg',
        amount : undefined
    },
    {
        category: categories[2],
        title: 'Crema Catalana',
        price: 5,
        description : 'At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.',
        path : 'img/dishes/crema_catalana.jpg',
        amount : undefined
    },
    {
        category: categories[2],
        title: 'CrÃªpe Suzette',
        price: 6,
        description : 'Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.',
        path : 'img/dishes/crepe.jpg',
        amount : undefined
    },
];