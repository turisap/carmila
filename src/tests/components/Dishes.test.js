/**
 * Created by HP on 15-Dec-17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import {Dishes} from '../../components/pages/Dishes'
import {dishes} from '../../data/data';

let wrapper, addItemToOrder;
beforeEach(() => {
    addItemToOrder = jest.fn();
    wrapper = shallow(<Dishes dishes={dishes} addItemToOrder={addItemToOrder}/>);
});

test('should render Dishes page', () => {
    expect(wrapper).toMatchSnapshot();
});

/* test('should add a dish to order', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(addItemToOrder).toHaveBeenCalledWith(dishes[0]);
});*/

