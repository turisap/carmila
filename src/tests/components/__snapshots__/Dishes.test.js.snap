// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dishes page 1`] = `undefined`;
